flink-runtime/src/test/java/org/apache/flink/runtime/testutils/MiniClusterResource.java

```diff
@@ -199,6 +200,9 @@ private void startMiniCluster() throws Exception {
                 new Configuration(miniClusterResourceConfiguration.getConfiguration());
         configuration.setString(
                 CoreOptions.TMP_DIRS, temporaryFolder.newFolder().getAbsolutePath());
+        configuration.set(
+                CheckpointingOptions.CHECKPOINTS_DIRECTORY,
+                temporaryFolder.newFolder().toURI().toString());
```

Comment:
Question, should we have this changed for everything using mini cluster, or only for our tests that are using unaligned checkpoints? 🤔 

Maybe for example because of benchmarks, and overall running times (costs of our CI), we should keep using memory storage if not needed?

Reply:
Unaligned checkpoint or change log state backend are enabled in the `TestStreamEnvironment#randomizeConfiguration`. And I see a large number of tests calling `FsStateChangelogStorageFactory.configure()` to Configure DFS DSTL in FLINK-23279, including: `StateChangelogOptions.STATE_CHANGE_LOG_STORAGE` and `FsStateChangelogOptions.BASE_PATH`. 

If we are worried about increasing the running time, we can set `CheckpointingOptions.CHECKPOINT_STORAGE` and `CheckpointingOptions.CHECKPOINTS_DIRECTORY` in a similar way, what do you think?

Reply:
FileSystemCheckpointStorage is compatible with more scenarios than JobManagerCheckpointStorage, for example: scenarios with medium or large state sizes. But its performance may be a little bit worse in small state scenarios.

I can fix the bug that the test fails, and then we look at the test duration. If there is no significant increase, `FileSystemCheckpointStorage` can be used as the default, what do you think?

Reply:
> If there is no significant increase, FileSystemCheckpointStorage can be used as the default, what do you think?

That might be ok. In that case we would also need to check the benchmark results.

@zentol had another suggestion to instead increase Akka's frame size and keep the memory storage?